### PR TITLE
Amqp messaging

### PR DESCRIPTION
**Pull request by nuzurie**

### CODE REVIEW

**Review by marksman1298 — APPROVED** [review]
Looks good. Great job Zubair

File: messaging/usecase/student_usecase.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+func failOnError(err error, msg string) {
+	if err != nil {
+		log.Fatalf("%s: %s", msg, err)
+	}
+}
+
+func (s studentUseCase) ListenStudentCreation() {
+	conn, err := amqp.Dial("amqp://guest:guest@localhost:5672/")
```
Comment (car12lin12) [inline messaging/usecase/student_usecase.go]: can you extract method for connection and channel creation for all listeners?

**Review by car12lin12 — APPROVED** [review]
great job as usual Zubair!!

File: messaging/usecase/student_usecase.go (hunk 1 above)
Comment (nuzurie) [reply messaging/usecase/student_usecase.go]: We need this urgently. We can refactor this later!

**Pull request merged**